dart/lib/src/hub.dart

```diff
@@ -510,20 +511,45 @@ class _StackItem {
 class _WeakMap {
   final _expando = Expando();
 
+  final SentryOptions _options;
+
+  _WeakMap(this._options);
+
   void add(
     dynamic throwable,
     ISentrySpan span,
     String transaction,
   ) {
-    if (throwable != null && _expando[throwable] == null) {
-      _expando[throwable] = MapEntry(span, transaction);
+    if (throwable == null) {
+      return;
+    }
+    try {
+      if (_expando[throwable] == null) {
+        _expando[throwable] = MapEntry(span, transaction);
+      }
+    } catch (exception, stackTrace) {
+      _options.logger(
+        SentryLevel.info,
+        'Throwable type: ${throwable.runtimeType} is not supported for associating errors to a transaction.',
```

Comment:
Maybe this should point to the [docs](https://api.dart.dev/stable/2.15.1/dart-core/Expando-class.html), which says which types are not supported? Or maybe include the list of unspported types?

Reply:
`throwable.runtimeType` does the job for now, but indeed a better logging message here would be awesome, I will merge and release for now because it's a bug and later improve it.